esp-hal/src/lcd_cam/lcd/i8080.rs

```diff
@@ -364,7 +381,48 @@ where
     }
 }
 
-impl<'d, CH: DmaChannel, P> I8080<'d, CH, P> {
+#[cfg(feature = "async")]
+impl<'d, CH: DmaChannel, P: TxPins> I8080<'d, CH, P, crate::Async>
+where
+    P::Word: Into<u16>,
+{
+    pub fn new_async(
+        lcd: Lcd<'d, crate::Async>,
+        channel: ChannelTx<'d, CH>,
+        descriptors: &'static mut [DmaDescriptor],
+        pins: P,
+        frequency: HertzU32,
+        config: Config,
+        clocks: &Clocks,
+    ) -> Self
+    where
+        <CH as crate::dma::DmaChannel>::P: crate::dma::LcdCamPeripheral,
+    {
+        I8080::new_internal(lcd, channel, descriptors, pins, frequency, config, clocks)
+    }
+
+    pub async fn send_dma_async<'t, TXBUF>(
+        &'t mut self,
+        cmd: impl Into<Command<P::Word>>,
+        dummy: u8,
+        data: &'t TXBUF,
+    ) -> Result<(), DmaError>
+    where
+        TXBUF: ReadBuffer<Word = P::Word>,
+    {
+        let (ptr, len) = unsafe { data.read_buffer() };
+
+        self.setup_send(cmd.into(), dummy);
+        self.start_write_bytes_dma(ptr as _, len * size_of::<P::Word>())?;
+        self.start_send();
+
+        DmaTxFuture::new(&mut self.tx_channel).await?;
```

Comment:
Might be safer to skip this. If the channel is in the wrong mode it'll hang forever.

If you just want to capture the error, if you check with `self.tx_channel.has_error()` have the lcd future returns.

Reply:
The question is if there was an error in the DMA descriptors would the lcd TX done interrupt fire.  If not then that would also hang forever.

Reply:
In reality the `Mode` should be extended to the TX and RX parts of the channel...

Reply:
Good question, in my experience playing around with this peripheral it won't wait for the DMA, so if something goes wrong with the DMA transfer, the peripheral will just send garbage then assert the interrupt, it won't hang.

Reply:
I would call that UB :-)

Reply:
I don't disagree :slightly_smiling_face:
Unfortunately this behavior would still occur regardless of whether the `DmaTxFuture` is used or not.

Reply:
I'll test this in a few days as I'm AFH (away from hardware) at the moment.

Reply:
> In reality the `Mode` should be extended to the TX and RX parts of the channel...I've not looked to see how much of a mess that would be. Random thought: maybe implement an into_async() and into_blocking()? Somehow make the tx and rx for chips that support it be able to be configured separately.

Ah sorry I missed this. I completely agree with this.
All the `DmaSupport` implementation will need to be updated, so a bit of faff but needed faff.
Possibly best as a separate PR but up to you.

Reply:
Yea that needs to be a seperate PR, I took your suggestion and tested the dma error after.